### ISSUE TITLE
Can't input Japanese on `docusaurus-theme-search-algolia` in Windows Firefox

### ISSUE THREAD

**Issue by tats-u** [facebook/docusaurus#8952]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

`docusaurus-theme-search-algolia` has a bug to prevent from inputting Japanese in Windows Firefox.
Other languages, e.g. Chinese, may be affected, too.
Chromium-based browsers work well.

### Reproducible demo

https://docusaurus.io/

### Steps to reproduce

1. Install or enable a Japanese input method (e.g. Google Japanese Input)
2. Click the search bar or Ctrl + K
3. Enable Japanese input method
4. Type some Japanese search query

### Expected behavior

https://user-images.githubusercontent.com/12870451/235352380-d6be7c95-c69b-489d-b8eb-d00d5ecdc499.mp4

(Edge)

Chrome also works well.

Note: this movie uses AV1. Safari may not be supported. Edge requires an [this Windows video extension](https://www.microsoft.com/store/productId/9MVZQVXJBQ9V).

### Actual behavior

https://user-images.githubusercontent.com/12870451/235352442-ccfbda41-a997-463d-8d56-26a10a3fd361.mp4

Input corrupted at every moment😡

### Your environment

- Public source code: officia pagel
- Public site URL: official page
- Docusaurus version used: official
- Environment name and version (e.g. Chrome 89, Node.js 16.4): N/A
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Windows + Firefox

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by tats-u** [comment]
https://developer.mozilla.org/docs/Web/API/Event/composed must be used to fix this problem.

**Comment by Josh-Cena** [comment]
I cannot reproduce this in macOS Firefox, but Chinese input methods work differently and I don't know how to use the Hiragana input method... In any case, Docusaurus does not provide the search modal, so we will track https://github.com/algolia/autocomplete/issues/1135 instead.

**Comment by tats-u** [comment]
@Josh-Ce I don't have a mac, but can't it be reproduced in Microsoft Pinyin in Windows parallels?

**Comment by Josh-Cena** [comment]
Unfortunately I don't have access to a Windows machine either :/

**Comment by tats-u** [comment]
Isn't it too early to close?

fyi how to enable Japanese IME in macOS

https://redcocoon.org/cab/mysoft.html

**Comment by Josh-Cena** [comment]
Yeah I do have the Hiragana keyboard, I just never know how to use it.

![Test](https://user-images.githubusercontent.com/55398995/235383366-afdbdaa5-a04c-4362-8e47-4535c3d8350a.gif)

**Comment by tats-u** [comment]
Seems to work well.
It may be Windows specific.

**Comment by tats-u** [comment]
I was able to test in Google Pinyin in Windows Sandbox.


https://user-images.githubusercontent.com/12870451/235466914-83505fb2-2bad-4a77-8191-75da32a24f49.mp4

Input sentence→Press backspace until all characters are deleted
While I pressed backspace, I found the search bar behaves suspiciously.

**Comment by tats-u** [comment]
Can macOS + Firefox input Pinyin at this field?

https://codepen.io/tats-u/pen/KKGXxEM

**Comment by Josh-Cena** [comment]
@tats-u sorry I was quite busy for the last few weeks. Yes, the pen works for me on Mac Firefox, both using Chinese and Hiragana keyboards.

The IME closing seems unexpected, but I can't reproduce it here. Maybe it's a platform-specific issue?

**Comment by tats-u** [comment]
macOS doesn't seem to be affected. Thank you for your cooperation.

FYI in Windows:

https://user-images.githubusercontent.com/12870451/235846510-b8fc90b6-2ee6-44e0-8a61-1d025bb58ec0.mp4

**Comment by tats-u** [comment]
https://github.com/algolia/autocomplete/issues/1135#issuecomment-1916879932

We will be able to fix this issue by [`ignoreCompositionEvents`](https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-ignorecompositionevents) in the latest autocomplete for CJK languages.

**Comment by Josh-Cena** [comment]
@tats-u Could you try it in your environment? Thanks!

**Comment by tats-u** [comment]
Has it been committed in main? It does not look like.

**Comment by Josh-Cena** [comment]
Ah, no, I meant if you wanted to make the code change and test it locally.

**Comment by tats-u** [comment]
I found we still wait for fix of https://github.com/algolia/docsearch/issues/1043 & https://github.com/algolia/docsearch/issues/1304.
That attribute has not been imported into `<DocSearchModal/>` (`@docsearch/react`) yet.

**Comment by Josh-Cena** [comment]
Is there a downside to this being unconditionally applied?

**Comment by tats-u** [comment]
You may want to see:

https://github.com/algolia/instantsearch/issues/5986
https://github.com/algolia/instantsearch/pull/6009

I am against their background, though. The detailed condition (the name of Ithe keyboard app & the language of the keyboard) is not written.
